Consider Trivia Content before multiline members - fixes #569

theimowski opened the pull request:
The pattern matching above uses nested `sepMember` function
to consider Trivia Content. The difference between those
pattern matching cases is that latter one matches on when
there are trailing non-multiline members.
There's some code duplication between those two now, which
I'm happy to refactor - feedback welcome.

nojaf commented on src/Fantomas.Tests/ClassTests.fs:
```diff
@@ -382,4 +382,19 @@ type Exception with
+"""
+
+[<Test>]
+let ``No extra new lines between type members, 569``() =
```
Start with a lowercase letter please.

theimowski commented:
Addressed the comments

I've also found 3 more cases where extra unwanted newline is being added - gonna tackle them as they're the outstanding issues preventing me from starting using fantomas in my project.
Already got 3 failing unit tests for them:

```fsharp
 [<Test>]
 let ``no extra new line before nested module with attribute, 569``() =
     let original = """module A =
     let x = 0

     [<RequireQualifiedAccess>]
     module B =
         let y = 1
 """

     formatSourceString false original config |> should equal original

 [<Test>]
 let ``no extra new line after elif expression, 569``() =
     let original = """let x =
     if true then printfn "a"
     elif true then printfn "b"

     if true then 1 else 0
 """

     formatSourceString false original config |> should equal original

 [<Test>]
 let ``no extra new line before abstract member with attribute, 569``() =
     let original = """type A =

     [<EmitConstructor>]
     abstract Create: Unit -> A

     abstract b: Unit -> Unit
 """

     formatSourceString false original config |> should equal original
```
Do you want me to combine those 3 more cases with this PR and issue, or shall I create a separate one?

nojaf commented:
Hey @theimowski , thanks for the changes.
Could you create a new issue of these other problems?
One would suffice for all three cases. 
Many thanks for these contributions!

theimowski commented:
Ok, created #586
